src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -1168,5 +1168,66 @@ public void NameofTypeParameterInParameterInitializer()
 }";
             var compilation = CreateCompilationWithMscorlib45(source).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(10467, "https://github.com/dotnet/roslyn/issues/10467")]
+        public void NameofFixedBuffer()
+        {
+            var source =
+@"
+using System;
+unsafe struct Struct1
+{
+  public fixed char MessageType[50];
```

Comment:
Since nameof is 100% compile-time, I think it should be safe.  I'll add tests for this.